tests/ai_hub/model_catalog/plugin_arch/utils.py

```
@@ -0,0 +1,46 @@
+            wait_timeout=timeout,
+            sleep=10,
+            func=requests.get,
+            url=url,
+            headers=headers,
+            verify=False,
+            timeout=10,
+        ):
+            if sample.status_code == expected_code:
+                return sample
+    except TimeoutExpiredError:
+        raise AssertionError(f"/readyz did not return {expected_code} within {timeout}s")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)